designer/server/src/main/scala/pl/touk/nussknacker/ui/api/description/DictApiEndpoints.scala

```diff
@@ -0,0 +1,142 @@
+package pl.touk.nussknacker.ui.api.description
+
+import io.circe.Json
+import io.circe.generic.JsonCodec
+import pl.touk.nussknacker.engine.api.dict.DictEntry
+import pl.touk.nussknacker.engine.api.process.ProcessingType
+import pl.touk.nussknacker.restmodel.BaseEndpointDefinitions
+import pl.touk.nussknacker.restmodel.BaseEndpointDefinitions.SecuredEndpoint
+import pl.touk.nussknacker.security.AuthCredentials
+import pl.touk.nussknacker.ui.api.description.DictApiEndpoints.DictError
+import pl.touk.nussknacker.ui.api.description.DictApiEndpoints.DictError.{
+  MalformedTypingResult,
+  NoDict,
+  NoProcessingType
+}
+import pl.touk.nussknacker.ui.api.description.DictApiEndpoints.Dtos._
+import sttp.model.StatusCode.{BadRequest, NotFound, Ok}
+import sttp.tapir._
+import sttp.tapir.json.circe.jsonBody
+
+import scala.language.implicitConversions
+
+class DictApiEndpoints(auth: EndpointInput[AuthCredentials]) extends BaseEndpointDefinitions {
+
+  lazy val dictionaryEntryQueryEndpoint: SecuredEndpoint[(String, String, String), DictError, List[DictEntry], Any] =
+    baseNuApiEndpoint
+      .summary("Get list of dictionary entries matching the label pattern")
+      .tag("Dictionary")
+      .get
+      .in(
+        "processDefinitionData" / path[String]("processingType") / "dicts" / path[String]("dictId") / "entry" / query[
+          String
+        ]("label")
+      )
+      .out(
+        statusCode(Ok).and(
+          jsonBody[List[DictEntry]]
+        )
+      )
+      .errorOut(
+        oneOf[DictError](
+          oneOfVariantFromMatchType(
+            NotFound,
+            plainBody[NoProcessingType]
+          ),
+          oneOfVariantFromMatchType(
+            NotFound,
+            plainBody[NoDict]
+          )
+        )
+      )
+      .withSecurity(auth)
+
+  lazy val dictionaryListEndpoint: SecuredEndpoint[(String, DictListRequestDto), DictError, List[DictDto], Any] =
+    baseNuApiEndpoint
+      .summary("Get list of available dictionaries with value type compatible with expected type")
+      .tag("Dictionary")
+      .post
+      .in("processDefinitionData" / path[String]("processingType") / "dicts")
+      .in(jsonBody[DictListRequestDto])
+      .out(
+        statusCode(Ok).and(
+          jsonBody[List[DictDto]]
+        )
+      )
+      .errorOut(
+        oneOf[DictError](
+          oneOfVariantFromMatchType(
+            NotFound,
+            plainBody[NoProcessingType]
+          ),
+          oneOfVariantFromMatchType(
+            BadRequest,
+            plainBody[MalformedTypingResult]
+          )
+        )
+      )
+      .withSecurity(auth)
+
+}
+
+object DictApiEndpoints {
+
+  object Dtos {
+
+    @JsonCodec
+    case class TypingResultInJson(value: Json)
+
+    @JsonCodec
+    case class DictListRequestDto(expectedType: TypingResultInJson)
+
+    @JsonCodec
+    case class DictDto(
+        id: String // TODO: we could introduce labels for dictionaries, currently we display them by id
```

Comment:
Please add the `label` field with the same value as `id` and move TODO to the place where we duplicate this. Thanks to that, FE will be ready for further improvement of this